src/pydvl/value/shapley/montecarlo.py

```diff
@@ -216,117 +362,81 @@ def permutation_montecarlo_shapley(
     See :ref:`data valuation` for details.
 
     :param u: Utility object with model, data, and scoring function.
-    :param n_iterations: total number of iterations (permutations) to use
-        across all jobs.
+    :param done: function checking whether computation must stop.
+    :param truncation: An optional callable which decides whether to
+        interrupt processing a permutation and set all subsequent marginals to
+        zero. Typically used to stop computation when the marginal is small.
     :param n_jobs: number of jobs across which to distribute the computation.
-    :param config: Object configuring parallel computation, with cluster address,
-        number of cpus, etc.
+    :param config: Object configuring parallel computation, with cluster
+        address, number of cpus, etc.
     :param progress: Whether to display progress bars for each job.
     :return: Object with the data values.
     """
-    iterations_per_job = max(1, n_iterations // n_jobs)
 
-    map_reduce_job: MapReduceJob[Utility, "NDArray"] = MapReduceJob(
+    map_reduce_job: MapReduceJob[Utility, ValuationResult] = MapReduceJob(
         u,
-        map_func=_permutation_montecarlo_marginals,
-        reduce_func=np.concatenate,  # type: ignore
-        map_kwargs=dict(max_permutations=iterations_per_job, progress=progress),
-        reduce_kwargs=dict(axis=0),
+        map_func=_permutation_montecarlo_shapley,
+        reduce_func=lambda results: reduce(operator.add, results),
```

Comment:
I find this much cleaner: all we need to do is add results together, and we get a valid result